downloadDirectory/src/test/java/org/reactome/release/downloadDirectory/GenerateGOAnnotationFile/CellularComponentAnnotationBuilderTest.java

```diff
@@ -54,39 +58,18 @@ public void cellularComponentAnnotationBuilderTest() throws Exception {
-        CellularComponentAnnotationBuilder.processCellularComponents(mockReactionInst);
-    }
-
-    @Test
-    public void cellularComponentInvalidProteinTest() throws Exception {
-        PowerMockito.mockStatic(GOAGeneratorUtilities.class);
-
-        mockProteinSet.add(mockProteinInst);
-        Mockito.when(GOAGeneratorUtilities.retrieveProteins(mockReactionInst)).thenReturn(mockProteinSet);
-        Mockito.when(mockProteinInst.getAttributeValue(ReactomeJavaConstants.referenceEntity)).thenReturn(mockReferenceEntityInst);
-        Mockito.when(mockProteinInst.getAttributeValue(ReactomeJavaConstants.species)).thenReturn(mockSpeciesInst);
-        Mockito.when((GOAGeneratorUtilities.isValidProtein(mockReferenceEntityInst, mockSpeciesInst))).thenReturn(false);
-        CellularComponentAnnotationBuilder.processCellularComponents(mockReactionInst);
-    }
-
-    @Test
-    public void cellularComponentExcludedMicrobialSpeciesTest() throws Exception {
-        PowerMockito.mockStatic(GOAGeneratorUtilities.class);
-
-        mockProteinSet.add(mockProteinInst);
-        Mockito.when(GOAGeneratorUtilities.retrieveProteins(mockReactionInst)).thenReturn(mockProteinSet);
-        Mockito.when(mockProteinInst.getAttributeValue(ReactomeJavaConstants.referenceEntity)).thenReturn(mockReferenceEntityInst);
-        Mockito.when(mockProteinInst.getAttributeValue(ReactomeJavaConstants.species)).thenReturn(mockSpeciesInst);
-        Mockito.when((GOAGeneratorUtilities.isValidProtein(mockReferenceEntityInst, mockSpeciesInst))).thenReturn(true);
-        Mockito.when(((GKInstance) mockSpeciesInst.getAttributeValue(ReactomeJavaConstants.crossReference))).thenReturn(mockCrossReferenceInst);
-        Mockito.when(mockCrossReferenceInst.getAttributeValue(ReactomeJavaConstants.identifier)).thenReturn("562");
-        Mockito.when(GOAGeneratorUtilities.isExcludedMicrobialSpecies("562")).thenReturn(true);
-        CellularComponentAnnotationBuilder.processCellularComponents(mockReactionInst);
+        Mockito.when(mockReferenceEntityInst.getAttributeValue(ReactomeJavaConstants.identifier)).thenReturn("R1234");
+        Mockito.when(GOAGeneratorUtilities.getSecondaryIdentifier(mockReferenceEntityInst)).thenReturn("R5678");
+        Mockito.when(GOAGeneratorUtilities.generateGOALine(mockReferenceEntityInst, GOAGeneratorConstants.CELLULAR_COMPONENT_LETTER, "GO:1234", "REACTOME:1234", GOAGeneratorConstants.TRACEABLE_AUTHOR_STATEMENT_CODE, "1234")).thenCallRealMethod();
+        List<String> goaLines = CellularComponentAnnotationBuilder.processCellularComponents(mockReactionInst);
+        assertThat(goaLines.size(), is(equalTo(1)));
+        assertThat(goaLines.get(0), is((equalTo("UniProtKB\tR1234\tR5678\t\tGO:1234\tREACTOME:1234\tTAS\t\tC\t\t\tprotein\ttaxon:1234"))));
     }
 
     @Test
-    public void cellularComponentAlternateGOCompartmentTest() throws Exception {
+    public void cellularComponentAlternateGOCompartmentReturnsZeroLinesTest() throws Exception {
```

Comment:
Nice -- I like this test!  Another one could be added to test the microbial species condition for excluding cellular components.

Reply:
Do you mean testing the excluded microbial species method? I have a test for that in the GOAGeneratorUtilities test if so

Reply:
Actually, I was thinking about testing this method directly with data that includes a microbial species and seeing that you get back what you expect for the generated lines.  Kind of a higher-level test that complements the existing test for the more focused utilities method.